R/footnote_plan.R

```diff
@@ -0,0 +1,96 @@
+#' Footnote Plan
+#'
+#' @param ...  footnote structure objects separated by commas
+#' @param marks type of marks required for footnotes. available options are "numbers", "letters", "standard" and "extended".
+#' The default option is set to "numeric".
+#'
+#' @return footnote plan object
+#' @export
+#'
+#' @examples
+#'
+#' footnote_plan <- footnote_plan(
+#'     footnote_structure(footnote_text = "Source Note"),
+#'     marks="letters")
+#'
+#' footnote_plan <- footnote_plan(
+#'     footnote_structure(footnote_text = "footnote", column_val = "Placebo"),
+#'     marks="numbers")
+#'
+footnote_plan <- function(...,marks=c("numbers","letters","standard","extended")){
+  footnote_structure_list <- list(...)
+  marks = match.arg(marks)
+
+  structure(
+    list(struct_list=footnote_structure_list, marks=marks),
+    class = c("footnote_plan", "plan")
+  )
+}
+
+#' Footnote Structure
+#'
+#' @param footnote_text string with text for footnote
+#' @param column_val string or a named list of strings which represent the column to apply the footnote to
+#' @param group_val string or a named list of strings which represent the value of group to apply the footnote to
+#' @param label_val string which represents the value of label to apply the footnote to
+#'
+#' @return footnote structure object
+#' @export
+#'
+#' @examples
+#'
+#' footnote_structure <- footnote_structure(footnote_text = "Source Note")
```

Comment:
It might be nice to add comments for each of these examples to understand what the result would be. Alternatively  you could add some details higher up about how the args can be used in combination or independently to target different cells